authority_manage/src/lib.rs

```diff
@@ -39,7 +39,9 @@ pub struct AuthorityManage {
 
 impl AuthorityManage {
     pub fn new() -> Self {
-        let logpath = ::std::env::var(DATA_PATH).expect(format!("{} must be set", DATA_PATH).as_str()) + "/authorities";
+        let logpath = ::std::env::var(DATA_PATH)
+            .expect(format!("{} must be set", DATA_PATH).as_str())
+            + "/authorities";
```

Comment:
I think `format!()` is more straightforward, because you can see the output string pattern directly.

Although `string + str` is a bit more efficient than `format!()` https://github.com/hoodie/concatenation_benchmarks-rs

Reply:
WTF?